homeassistant/helpers/update_coordinator.py

```diff
@@ -61,6 +62,10 @@ def __init__(
         # when it was already checked during setup.
         self.data: _T = None  # type: ignore[assignment]
 
+        # Pick a random microsecond to stagger the refreshes
+        # and avoid a thundering herd.
+        self._microsecond = randint(50000, 500000)
```

Comment:
I picked a minimum of 50000 microseconds to avoid aligning things to 0 since there are likely other places that happens given track_time_interval also has this issue since `dt_util.find_next_time_expression_time` zeros out microseconds

Reply:
I guess I should fix that as well

Reply:
I just happen to have enough coordinators to get it to happen.  Someone else is going to have enough interval trackers